network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/pixels/PrivateDnsSettingReporter.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.pixels
+
+import android.content.Context
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.app.global.extensions.getPrivateDnsServerName
+import com.duckduckgo.app.global.extensions.isPrivateDnsActive
+import com.duckduckgo.di.scopes.VpnScope
+import com.duckduckgo.mobile.android.vpn.service.VpnServiceCallbacks
+import com.duckduckgo.mobile.android.vpn.state.VpnStateMonitor.VpnStopReason
+import com.duckduckgo.networkprotection.api.NetworkProtectionState
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@ContributesMultibinding(VpnScope::class)
+class PrivateDnsSettingReporter @Inject constructor(
+    private val networkProtectionPixels: NetworkProtectionPixels,
+    private val dispatcherProvider: DispatcherProvider,
+    private val networkProtectionState: NetworkProtectionState,
+    private val context: Context,
+) : VpnServiceCallbacks {
+    override fun onVpnStarted(coroutineScope: CoroutineScope) {
+        coroutineScope.launch {
+            reportPrivateDns(false)
```

Comment:
This will be reported for when AppTP only users as well. I know that this is a VPN specific setting which could affect AppTP and NetP, but if we want to measure how much of NetP DAUs have private DNS enabled, then it would make sense to emit thie for Netp users only?

Reply:
`reportPrivateDns` has an `if(networkProtectionState.isEnabled()) {...}`.
Did I miss something?

Reply:
Oh I did miss it. Should we also call this in `onVPNReconfigured` then?